src/components/OnboardingManager.tsx

```diff
@@ -0,0 +1,133 @@
+'use client';
+
+import {
+  BackLink,
+  FormDiv,
+  OuterDiv,
+  FormContainer,
+} from '@/app/onboarding/styles';
+import { P } from '@/styles/text';
+import { OnboardingContext } from '@/utils/OnboardingProvider';
+import { usePathname, useRouter } from 'next/navigation';
+import { ReactNode, useContext, useEffect, useMemo } from 'react';
+import ProgressBar from './ProgressBar';
+import BigButton from './BigButton';
+
+export default function OnboardingManager({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const onboarding = useContext(OnboardingContext);
+  const router = useRouter();
+  const pathname = usePathname();
+  const pageProgress = useMemo(() => {
+    if (!onboarding) return 0;
+    const find = onboarding.flow.findIndex(
+      f => `/onboarding/${f.url}` === pathname,
+    );
+    return find !== -1 ? find : 0;
+  }, [pathname, onboarding]);
+
+  useEffect(() => {
+    if (!onboarding) return;
+
+    // out of bounds redirect
+    if (
```

Comment:
what are the chances that this happens? is it only if the user manually routes to an index that is out of bounds